lumen_runtime/src/otp/maps/keys_1/test/with_map.rs

```diff
@@ -0,0 +1,26 @@
+use super::*;
+
+use liblumen_alloc::erts::term::atom_unchecked;
+
+#[test]
+fn returns_list_of_keys() {
```

Comment:
I'm not a huge fan of this property test. I think that ideally I'd be generating a list of terms, and then using that list to construct the map.

I was running into two issues with that approach though:

1) Since `:maps.keys/1` doesn't guarantee any ordering, I wasn't sure of what the most idiomatic way would be to compare the two lists at the end (since just doing a straight equality check may be flaky)
2) I'm not clear yet on the cleanest ways of manipulating maps / lists / terms in general from Rust, and specifically how I'd actually create the map from a list of keys.

If you have any advice on making this test more thorough, I'm happy to hear it :)

Reply:
🤷‍♂  this is good enough.  I've also not had a good approach for these order independent invariants either.  You could use `proptest::collection::vec(strategy::term(), 0..3)` to get a vector of 0-3 terms and then sort both the vector and the keys that are output to make sure they match, but it is fine without that.

If you're not going to go that complex, I would add an additional test that checks that a map with no keys returns an empty list (`Term::NIL`) as that is an important edge case.